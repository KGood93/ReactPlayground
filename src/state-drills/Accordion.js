import React from 'react';

class Accordion extends React.Component {
    static defaultProps = {sections: []};

    state = {
        currentSection: null,
    };
    
    handleButtonClick = (index) => {
        this.setState({ currentSection: index })
    }

    renderItems(section, idx, currentSection) {
        return (
            <li className='item' key={idx}>
                <button type='button' onClick={() => this.handleButtonClick(idx)}>
                    {section.title}
                </button>
                {(currentSection === idx) && <p>{section.content}</p>}
            </li>
        )
    }
    
    render() {
        const {currentSection} = this.state
        const {sections} = this.props
        return (
            <ul className='Accordion'>
                {sections.map((section,idx) => this.renderItems(section, idx, currentSection)
                )}
            </ul>
        )
  }  
}

export default Accordion;