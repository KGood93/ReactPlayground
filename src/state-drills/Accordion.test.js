import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const sections = [
    { title: 'Section 1',
        content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque.'},
    { title: 'Section 2',
        content: 'Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit.'},
    { title: 'Section 3',
        content: 'Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur.'},
  ];

describe(`Accordion Component`, () => {
    it('renders empty given no Accoridon without errors', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders no sections by default', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('opens any clicked section', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('only opens one section at a time', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})