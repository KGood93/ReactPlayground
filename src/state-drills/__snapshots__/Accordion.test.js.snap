// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component renders empty given no Accoridon without errors 1`] = `
<ul
  className="Accordion"
/>
`;
